core/src/main/java/org/infinispan/expiration/impl/ExpirationManagerImpl.java

```diff
@@ -269,11 +261,11 @@ private void deleteFromStoresAndNotify(K key, V value, Metadata metadata, Privat
     * or not, that is if it couldn't be touched - we assumed expired (as it was removed in some way).
     * @param entry the entry to check expiration and touch
     * @param segment the segment the entry maps to
+    * @param currentTime the current time in milliseconds
     * @return whether the entry was expired or not
     */
-   protected CompletionStage<Boolean> checkExpiredMaxIdle(InternalCacheEntry entry, int segment) {
-      TouchCommand touchCommand = cf.running().buildTouchCommand(entry.getKey(), segment);
-      CompletionStage<Boolean> future = (CompletionStage) touchCommand.invokeAsync(componentRegistry);
+   protected CompletionStage<Boolean> checkExpiredMaxIdle(InternalCacheEntry entry, int segment, long currentTime) {
+      CompletionStage<Boolean> future = cache.touch(entry.getKey(), segment, true);
```

Comment:
shouldn't `touchEvenIfExpired` be  `false` here?

Reply:
No, it is to cover the edge case of the entry expiring between when we check and then actually touching. We just don't want to touch for things like the xsite max idle check.